lib/puppet-lint.rb

```diff
@@ -180,9 +180,8 @@ def report(problems)
         print_github_annotation(message) if configuration.github_actions
       end
     end
-    puts JSON.pretty_generate(json) if configuration.json
```

Comment:
This could affect usage via Rake. I don't know if people do this, but there are effectively 2 entry points: the bin file which you already found and Rake:
https://github.com/puppetlabs/puppet-lint/blob/020143b705b023946739eb44e7c7d99fcd087527/lib/puppet-lint/tasks/puppet-lint.rb#L88-L107=

Reply:
thanks for all the inputs! I will set the PR as draft for now and look into them.